pkg/providers/ovirt/provider.go

```diff
@@ -233,17 +233,27 @@ func (o *OvirtProvider) ProcessTemplate(template *templatev1.Template, vmName *s
 	if err != nil {
 		return nil, err
 	}
-	labels, err := o.templateFinder.GetLabels(sourceVM)
+	labels, annotations, err := o.templateFinder.GetLabels(template, sourceVM)
 	if err != nil {
 		return nil, err
 	}
-	return updateLabels(vm, labels), nil
+	updateLabels(vm, labels)
+	updateAnnotations(vm, annotations)
+	return vm, nil
 }
 
-func updateLabels(vm *kubevirtv1.VirtualMachine, labels map[string]string) *kubevirtv1.VirtualMachine {
+func updateLabels(vm *kubevirtv1.VirtualMachine, labels map[string]string) {
 	utils.AddLabels(vm.ObjectMeta.GetLabels(), labels)
 	utils.AddLabels(vm.Spec.Template.ObjectMeta.GetLabels(), labels)
-	return vm
+}
+
+func updateAnnotations(vm *kubevirtv1.VirtualMachine, annotationMap map[string]string) {
+	anotations := vm.ObjectMeta.GetAnnotations()
+	if anotations == nil {
+		anotations = make(map[string]string)
+		vm.ObjectMeta.SetAnnotations(anotations)
+	}
+	utils.AddLabels(anotations, annotationMap)
```

Comment:
this is a bit confusing, since the name is addLabels but it deals with annotations, perhaps `utils.AddLabels` needs to be generalized : "AppendMap"/ "AddCollection" / "AddToMap"

Reply:
ack